### ISSUE TITLE
the training of VQ-VAE is ineffective

### ISSUE THREAD

**Issue by Sun-Happy-YKX** [Youncy-Hu/MAGE#5]
the training of VQ-VAE is ineffective.
I have download "Moving MNIST datasets" by the provided code, the final folder is as follows.
![image](https://user-images.githubusercontent.com/50354191/236680272-d5fd9a42-85ba-4f6c-b82a-3a4201ea9c03.png)
But when I execute the VQ-VAE training code, the loss of test does not reduce, it keep stable at a very high level as follow.
![image](https://user-images.githubusercontent.com/50354191/236680344-fdef4c45-60a8-4dbc-bb92-3241624ac097.png)
I would like you to give me some help, thanks.

**Comment by Sun-Happy-YKX** [comment]
![image](https://user-images.githubusercontent.com/50354191/236756331-16540686-79eb-424b-96c1-e4974ece9cf0.png)
a mistake occurs when I begin to train, add it for supplement.

**Comment by Sun-Happy-YKX** [comment]
![image](https://user-images.githubusercontent.com/50354191/236773067-10aa3a91-502a-40b4-8e48-667a43c52cf3.png)
I have annoted these codes about "ckpt_path" last training, I think it is the reason of above problem. 
![image](https://user-images.githubusercontent.com/50354191/236773707-4f35e2f4-b512-4613-8eff-2f94a2bbd11c.png)
but the "ckpt_path" is not provided, could you please give some indications.

**Comment by Sun-Happy-YKX** [comment]
@Youncy-Hu

**Comment by Sun-Happy-YKX** [comment]
![image](https://user-images.githubusercontent.com/50354191/236810734-88e818e1-2935-41c6-b28b-bdc4fce1a22d.png)
I observe that the quantization loss is expectable, but the reconstruction loss does not decrease any more.
@Youncy-Hu

**Comment by wxrui182** [comment]
Hello, I also encountered the same problem. The reconstruction_loss is always very high during the training process of VQ-VAE. Have you solved this problem?

**Comment by wxrui182** [comment]
> the training of VQ-VAE is ineffective. I have download "Moving MNIST datasets" by the provided code, the final folder is as follows. ![image](https://user-images.githubusercontent.com/50354191/236680272-d5fd9a42-85ba-4f6c-b82a-3a4201ea9c03.png) But when I execute the VQ-VAE training code, the loss of test does not reduce, it keep stable at a very high level as follow. ![image](https://user-images.githubusercontent.com/50354191/236680344-fdef4c45-60a8-4dbc-bb92-3241624ac097.png) I would like you to give me some help, thanks.

Hello, the problem lies in this line of code: "train_dataset = MNIST4VQVAE(args.data_root, 'train', image_transform=transform)". Just change "image_transform=transform" to "image_transform=None". The same is true for "test_dataset = MNIST4VQVAE(args.data_root, 'test', image_transform=None)".

**Comment by wxrui182** [comment]
> > the training of VQ-VAE is ineffective. I have download "Moving MNIST datasets" by the provided code, the final folder is as follows. ![image](https://user-images.githubusercontent.com/50354191/236680272-d5fd9a42-85ba-4f6c-b82a-3a4201ea9c03.png) But when I execute the VQ-VAE training code, the loss of test does not reduce, it keep stable at a very high level as follow. ![image](https://user-images.githubusercontent.com/50354191/236680344-fdef4c45-60a8-4dbc-bb92-3241624ac097.png) I would like you to give me some help, thanks.
> 
> Hello, the problem lies in this line of code: "train_dataset = MNIST4VQVAE(args.data_root, 'train', image_transform=transform)". Just change "image_transform=transform" to "image_transform=None". The same is true for "test_dataset = MNIST4VQVAE(args.data_root, 'test', image_transform=None)".
![reconstruction](https://github.com/Youncy-Hu/MAGE/assets/56919418/71516649-acd8-412b-9846-2f55c6179f21)
![loss](https://github.com/Youncy-Hu/MAGE/assets/56919418/9c54f9b7-c959-4dc3-bc84-2f291c644ee3)

**Comment by wxrui182** [comment]
> ![image](https://user-images.githubusercontent.com/50354191/236773067-10aa3a91-502a-40b4-8e48-667a43c52cf3.png) I have annoted these codes about "ckpt_path" last training, I think it is the reason of above problem. ![image](https://user-images.githubusercontent.com/50354191/236773707-4f35e2f4-b512-4613-8eff-2f94a2bbd11c.png) but the "ckpt_path" is not provided, could you please give some indications.

Follow the code I modified above to train vqvae and save the model weights. Just load the weights of vqvae at this location.

**Comment by wenyufeng936** [comment]
Hello, have you reproduced the results of the paper? Also, how much video memory is needed to run this experiment? Can you please let me know? thank you
